cmd/celestia-da/light.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	cmdnode "github.com/celestiaorg/celestia-node/cmd"
+	"github.com/celestiaorg/celestia-node/nodebuilder/core"
+	"github.com/celestiaorg/celestia-node/nodebuilder/gateway"
+	"github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/nodebuilder/p2p"
+	"github.com/celestiaorg/celestia-node/nodebuilder/rpc"
+	"github.com/celestiaorg/celestia-node/nodebuilder/state"
+)
+
+// NOTE: We should always ensure that the added Flags below are parsed somewhere, like in the
+// PersistentPreRun func on parent command.
+
+func init() {
```

Comment:
By doing this in init, you end up having 3 servers for every node type, no matter which one is run. Besides this being a bad design, this might also not work because listening on the same port and address

Reply:
The server is only called when `start` command is run. This seems to be work:

`docker run opcelestia/celestia-da:v0.12.0 celestia-da light start --grpc.namespace 000008e5f679bf7116cb --grpc.token xxx`

I had to use this roundabout way because node doesn't export the main command. So there's no way to access and inject the server into the `Start` subcommand.

Reply:
Suggested a feature request to node which would make this a lot clear to implement: https://github.com/celestiaorg/celestia-node/issues/2936

Reply:
Implemented 🤝